File: 13405-491-118-pascals-triangle/13405-491-118-pascals-triangle.ts
function generate(numRows: number): number[][] {
    let result: number[][] = [
        [1],
    ]

    if (numRows  == 1) {
        return result
    } else {
        result.push([1,1])
    }

    if (numRows === 2) {
        return result;
    }

    while (true) {
        let currLength : number = result.length;
        let prevRow : number[] = result[currLength - 1];
        let res : number[] = [];

        for (let i : number = 0; i < prevRow.length - 1; i++) {
            let resCalculate = calculate(prevRow[i], prevRow[i + 1])
            res.push(resCalculate)
        }
        result.push([1, ...res ,1])

        if (result.length == numRows) break;
    }
    return result
};

function calculate(numb1: number, numb2: number): number {
    return numb1 + numb2
}

File: 13405-491-118-pascals-triangle/13405-491-118-pascals-triangle.go
func generate(numRows int) [][]int {
    result := [][]int{{1}}
    
    if numRows == 1 {
        return result
    } 
    
    result = append(result, []int{1,1})
    
    for len(result) < numRows {
        currLength := len(result)
        prevRow := result[currLength -1]
        res := []int{1}
        
        for i := 0; i < len(prevRow) -1; i++{
            res = append(res, calculate(prevRow[i], prevRow[i+1]))
        }
        
        res = append(res, 1)
        
        result = append(result, res)
    }
    
    return result
}

func calculate(n1, n2 int) int {
    return n1 + n2
}